aten/src/ATen/native/cuda/ScatterGatherKernel.cu

```diff
@@ -137,19 +133,6 @@ struct cuda_scatter_gather_base_kernel {
     const std::string& method_name,
     const func_t& f
   ) {
-    // no-op if index is empty
-    if (index.numel() == 0) {
-      return;
-    }
-    at::assert_no_internal_overlap(self);
```

Comment:
I think we still need this assertion here.

Reply:
You are right, it's now fixed.